File: anul2/semestrul1/map/lab2/model/Task.java
package model;

import java.util.Objects;

public abstract class Task {
    private String taskID;
    private String descriere;

    public Task(String taskID, String descriere) {
        this.taskID = taskID;
        this.descriere = descriere;
    }

    public String getTaskID() {
        return taskID;
    }

    public void setTaskID(String taskID) {
        this.taskID = taskID;
    }

    public String getDescriere() {
        return descriere;
    }

    public void setDescriere(String descriere) {
        this.descriere = descriere;
    }

    //metoda abstracta
    public abstract void execute();

    @Override
    public String toString() {
        return "Task{" +
                "taskID='" + taskID + '\'' +
                ", descriere='" + descriere + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Task task = (Task) o;
        return Objects.equals(taskID, task.taskID) && Objects.equals(descriere, task.descriere);
    }

    @Override
    public int hashCode() {
        return Objects.hash(taskID, descriere);
    }
}

File: Third Semester - Advanced Programming Methods/Lab10/model/Task.cs
namespace lab10.model;

public abstract class Task
{
    private string _taskId;
    private string _description;

    public Task(string taskId, string description)
    {
        this._taskId = taskId;
        this._description = description;
    }
    
    public string TaskID { get => _taskId; set => _taskId = value; }
    
    public string Description { get => _description; set => _description = value; }
    
    public override string ToString()
    {
        return $"Task: {_description}\nTaskID: {_taskId}";
    }

    public override bool Equals(object? obj)
    {
        if (this == obj)
            return true;
        if (obj == null || GetType() != obj.GetType())
            return false;

        Task task = (Task)obj;
        return Equals(_taskId, task._taskId) && Equals(_description, task._description);

    }

    public override int GetHashCode()
    {
        return HashCode.Combine(_taskId, _description);
    }

    public abstract void execute();
}